Fix: Tag removal and navigation conflicts on Contact page

DarshBhavnani opened the pull request:
## Problem

Tag interactions on Contact pages have conflicting behaviors:
- Clicking a tag text removes the tag instead of viewing it
- Clicking the X button redirects to tag view page instead of just removing the tag

## Solution

Separate tag navigation from tag removal actions using `dismissible_onclick` attribute:
- Tag text now only navigates to tag view page
- X button now only removes the tag without page redirect

## Steps to Reproduce

### Before Fix:
1. Go to any Contact with tags
2. Click on tag text → **BUG**: Tag gets removed + redirects to tag page
3. Click the X button on a tag → **BUG**: Tag gets removed + redirects to tag page

### After Fix:
1. Go to any Contact with tags  
2. Click on tag text → ✅ Only redirects to tag view page
3. Click the X button → ✅ Only removes the tag (no redirect)

## Technical Changes

- **Template**: Use `dismissible_onclick` instead of `onclick` in `_tag.html.twig`
- **JavaScript**: Fix icon targeting and event handling in `removeTagFromLead()`
- **Structure**: Move dismissible actions outside of `attributes` object for better organization

Fixes #15311

DarshBhavnani committed:
fix: Prevent tag removal on link click

Signed-off-by: Darsh Bhavnani <darshbhavnani@gmail.com>

DarshBhavnani committed:
fix: Prevent tag removal button from triggering parent link

Signed-off-by: Darsh Bhavnani <darshbhavnani@gmail.com>

kuzmany commented on app/bundles/LeadBundle/Resources/views/Lead/lead.html.twig:
```diff
@@ -551,10 +551,10 @@
                                     },
                                     'color': 'blue',
                                     'type': 'dismissible',
+                                    'onclick': 'Mautic.removeTagFromLead(this, ' ~ lead.id ~ ', ' ~ tag.id ~ ', event);'
```
Why we need this change?

DarshBhavnani replied:
Earlier, this event handler was passed in attributes https://github.com/mautic/mautic/blob/c951c3606f8f316219e5646c78ad683f1f6b26b6/app/bundles/LeadBundle/Resources/views/Lead/lead.html.twig#L551-L555
 https://github.com/mautic/mautic/blob/c951c3606f8f316219e5646c78ad683f1f6b26b6/app/bundles/CoreBundle/Resources/views/Helper/_tag.html.twig#L47-L49
The event handler was then being applied to wrapper tag i.e **a** tag in this case!
To solve the error I removed the handler from attributes

kuzmany replied:
But I was able to do same with 

```
 'attributes': { 
     'href': path('mautic_tagmanager_action', {'objectAction': 'view', 'objectId': tag.id}), 
     'data-toggle': 'ajax', 
	 'onclick': 'Mautic.removeTagFromLead(this, ' ~ lead.id ~ ', ' ~ tag.id ~ ', event);'
 }, 
```

Then asking why you made the change.

DarshBhavnani replied:
All the attributes here will be applied to wrapper tag! But I think event listener should be applied to button and not the tag! So I moved it!

I did solve it without moving but I thought this was correct way to solve it!
If you want, I can move it back!

kuzmany committed:
Improve tag interaction: Use dismissible_onclick for better separation

- Replace 'onclick' with 'dismissible_onclick' attribute in _tag.html.twig
- Move dismissible_onclick outside of attributes object for better structure
- Update all tag instances in lead.html.twig to use new attribute structure
- Fix JavaScript function to properly target icon element for loading animation
- Ensure consistent event parameter passing for all tag removal calls

This provides cleaner separation between tag navigation (href) and dismissal (dismissible_onclick) actions.

kuzmany commented:
@DarshBhavnani I rename attribute to dismissible_onclick to be clear what does real do.
Also fixed it in more places and fixed JS finder

Now should work properly

DarshBhavnani commented:
Looks great, @kuzmany . Thanks so much for the quick fix and for getting this sorted!

kuzmany reviewed (approved):
I was able to reproduce and fix small inconsistence.
Works as expected now.

LeonOltmanns reviewed (approved):
Nice catch and good fix. :)

matbcvo commented:
@all-contributors please add @DarshBhavnani for code
